plugins/scaffolder-backend/src/service/router.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+import { Logger } from 'winston';
+import Router from 'express-promise-router';
+import express from 'express';
+import { StorageBase, TemplaterBase } from '../scaffolder';
+
+export interface RouterOptions {
+  storage: StorageBase;
+  templater: TemplaterBase,
+  logger: Logger;
+}
+
+export async function createRouter(
+  options: RouterOptions,
+): Promise<express.Router> {
+  const router = Router();
+  const {storage, templater, logger: parentLogger} = options;
+  const logger = parentLogger.child({ plugin: 'scaffolder' });
+
+  router
+    .get('/v1/templates', async (_, res) => {
+      const templates = await storage.list();
+      res.status(200).json(templates);
+    }).post('/v1/job/create', async (_, res) => {
+      // TODO(blam): Actually make this function work
+      const mock = 'templateid';
+      res.status(201).json({ accepted: true });
+
+      const path = await storage.prepare(mock);
+      await templater.run({ directory: path, values: { componentId: 'test' } });
+  });
+
+  const app = express();
+  app.set('logger', logger);
```

Comment:
@Rugvip wanna repeat what your idea was for doing this differently? i didn't understand last time

Reply:
As it's done in the rest of this PR. This isn't used atm is it?

Reply:
Ok you said something about middlewares. Yeah, this can currently be removed. But would it make sense to want per-request loggers? To get trace ids and similar things logged along with whatever happens during the request cycle.

Reply:
(of course that doesn't have to be solved immediately)

Reply:
Yeah I've had a middleware for decorating the request with a `logger` that includes trace IDs in the past. Might be something we should move to early on to make it easier to use something like https://github.com/opentracing-contrib/javascript-express

Reply:
https://github.com/skonves/express-http-context

Hmmmmmmm to set a logger and stuff on

Reply:
@freben that library confuses me. Looking at the library I have no idea how it can deal with async functions and requests as the scope changes for every request. so surely the logger would update for every request too?

Or have I misunderstood how that library works?

Reply:
+1 for open tracing though. Would be good to dogfood backstage through backstage so we can have a tracing plugin to see how everything fits together.

Reply:
I believe it uses V8 functionality to track a context across promises and callbacks. Essentially, V8 seems to have hooks that let you do actions on js scopes as they get crested, so you can tie them to the parent context. Or something like that

Reply:
But this was a note for the future, don't block on it. Merge what you have, or remove this extra express wrapping both here and in the inventory plugin. Either is fine with me

Reply:
(Oh and reading your comment again - yes, there would indeed intentionally be a new logger child created at the start of every request)

Reply:
@freben - ah, just did some reading about `cls-hooked` the library that it uses underneath and found this writeup, which turns out to be the library that we're using
https://medium.com/trabe/continuation-local-storage-for-easy-context-passing-in-node-js-2461c2120284